elastic/kibana
# [Infrastructure UI] Fix inventory table pagination navigation

## 📓 Summary

Closes #153739 

Each time the whole inventory table was rerendering due to a click on an entry, a new list of items was regenerated, reinitializing the in-memory table component from the first page.

Memoizing the generated items solves the issue and the table work as expected.

This PR also solve another issue occurring when multiple table entries were clicked, which resulted in many popovers opened and a blocked UI as per the following screenshot:

<img width="1267" alt="multi-popover" src="https://user-images.githubusercontent.com/34506779/228196645-9c0444b7-9114-47c2-8e5d-0bec29ed5305.png">

## 🧪 Testing
- Navigate to `Inventory` and select the `Kubernetes Pods` option from the  **Show** filter.
- Switch the view to a table with the top-right selector
- Verify the popover correctly opens on any entry of the first table page, then switch to different pages and check the behaviour is maintained and there are no reinitializations for the table.
- Verify the same from the `Docker Containers` list.


https://user-images.githubusercontent.com/34506779/228198613-afafe3fe-f714-4c88-a288-fc4c95ca801a.mov
